feat: updated scores retaining logic and weights setting intervals

daemontplx opened the pull request.

jarvis8x7b commented on commons/human_feedback/dojo.py:
```diff
@@ -95,89 +116,92 @@ async def create_task(
+        for attempt in range(max_retries):
+            try:
+                path = f"{DOJO_API_BASE_URL}/api/v1/tasks/create-tasks"
+                taskData = cls.serialize_feedback_request(feedback_request)
```
can use snake case

Tedbees commented:
:tada: This PR is included in version 1.4.2 :tada:

The release is available on [GitHub release](https://github.com/tensorplex-labs/dojo/releases/tag/v1.4.2)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

daemontplx committed:
chore: updated logging for scoring

daemontplx committed:
fix: clamp negative weights to 0

tensorplex-validator committed:
Merge pull request #68 from tensorplex-labs/chore/scores-logging

chore: updated logging for scoring, fixed negative weights

tensorplex-validator committed:
chore: added subtensor and watchtower update

tensorplex-validator committed:
chore: fix the name typo in networks specification

tensorplex-validator committed:
chore: specify local subtensor setup as optional

tensorplex-validator committed:
Merge pull request #70 from tensorplex-labs/chore/added-more-docker-stuffs

chore: added subtensor and watchtower update

tensorplex-validator committed:
chore: specifying watchtower config (#72)

* chore: specifying a startup configuraiton for watchtower

* chore: removed networks label in watchtower compose

tensorplex-validator committed:
Merge branch 'main' into dev

daemontplx committed:
feat: updated scores retaining logic and weights setting intervals (#75)

* feat: updated scores retaining logic for weights setting

scores are only reset based on ScoreResetInterval and weights are
continously set based on WeightSettingInterval based on retained scores
of that period

* chore: reduced logging level for deregistered hotkeys when calculating score

* feat: used a rolling window method to calculate scores instead

* feat: updated rolling window expired votes api endpoint

* feat updated route, no longer updates task status to scored
